staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -659,16 +654,22 @@ func validateCustomResourceDefinitionValidation(customResourceValidation *apiext
 			allowDefaults:            opts.allowDefaults,
 			requireValidPropertyType: opts.requireValidPropertyType,
 		}
+
 		allErrs = append(allErrs, ValidateCustomResourceDefinitionOpenAPISchema(schema, fldPath.Child("openAPIV3Schema"), openAPIV3Schema, true)...)
 
-		if opts.requireStructuralSchema {
+		if opts.requireStructuralSchema && len(allErrs) == 0 {
 			if ss, err := structuralschema.NewStructural(schema); err != nil {
 				// if the generic schema validation did its job, we should never get an error here. Hence, we hide it if there are validation errors already.
-				if len(allErrs) == 0 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("openAPIV3Schema"), "", err.Error()))
+			} else {
+				if errs := structuralschema.ValidateStructural(ss, fldPath.Child("openAPIV3Schema")); len(errs) > 0 {
```

Comment:
I'm in favor of this change (require structural to pass before checking defaulting)